src/Core/Components/DataGrid/FluentDataGridCell.razor.cs

```diff
@@ -80,7 +80,7 @@ public partial class FluentDataGridCell<TGridItem> : FluentComponentBase
         .AddStyle("height", "100%", Grid.MultiLine)
         .AddStyle("min-height", "44px", Owner.RowType != DataGridRowType.Default)
         .AddStyle("display", "flex", ShouldHaveDisplayFlex())
-        .AddStyle("z-index", (Grid._columns.Count + 1 - Column?.Index).ToString(), CellType == DataGridCellType.ColumnHeader && Grid._columns.Count > 0)
+        .AddStyle("z-index", "1", CellType == DataGridCellType.ColumnHeader && Grid._columns.Count > 0)
```

Comment:
You could create a new value in the `ZIndex.cs` class to centralize these values